pkg/controllers/user/certsexpiration/certsexpiration.go

```diff
@@ -67,6 +72,9 @@ func (c Controller) sync(key string, cluster *v3.Cluster) (runtime.Object, error
 		}
 		certsExpInfo[certName] = info
 	}
+	logrus.Debugf("Checking and deleting unused kubelet certificates, current nodes are: %v", hosts.NodesToHosts(cluster.Status.AppliedSpec.RancherKubernetesEngineConfig.Nodes, ""))
```

Comment:
These log messages are going to be massive with anything other than a single node setup. We can't log all nodes in a message even under debug. If you need a debug make it specific to something useful like `deleting certs for cluster x`.

Reply:
This can also leak security info related to nodes.